election-server/src/bin/indexer.rs

```diff
@@ -82,6 +105,80 @@ pub struct BlockData {
     pub ballots:    Vec<BallotSubmission>,
 }
 
+/// The contextual parameters necessary to generate the encrypted ballot
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct ElectionGuardConfig {
+    /// The election manifest. This should be declared externally for each
+    /// election.
+    pub election_manifest:    ElectionManifest,
+    /// The election parameters. These should be generated externally for each
+    /// election.
+    pub election_parameters:  ElectionParameters,
+    /// The guardian public keys, which are registered in the election contract.
+    pub guardian_public_keys: Vec<GuardianPublicKey>,
+}
+
+impl TryFrom<&AppConfig> for ElectionGuardConfig {
+    type Error = anyhow::Error;
```

Comment:
This is a bit nitpicky, but TryFrom I think should be pure functions typically, taht may fail. Reading from a file goes against that.

For parsing from files or reading from a filesystem I think there should be explicit methods on the relevant struct, in this case `AppConfig`.

Reply:
I agree, I also felt a bit dirty when I did it, so happy to change!